h2o-core/src/main/java/water/parser/ParseDataset.java

```diff
@@ -968,17 +990,24 @@ private FVecParseWriter streamParse(final InputStream is, final ParseSetup local
           break;
         }
         if ((_setup.getParseType().name().toLowerCase().equals("svmlight") ||
-                (_setup.getParseType().name().toLowerCase().equals("avro") ))
-                && ((_setup.getSkippedColumns() != null) && (_setup.getSkippedColumns().length >0)))
+                (_setup.getParseType().name().toLowerCase().equals("avro")))
+                && ((_setup.getSkippedColumns() != null) && (_setup.getSkippedColumns().length > 0)))
           throw new H2OIllegalArgumentException("Parser: skipped_columns are not supported for " +
                   "SVMlight or Avro parsers.");
 
-        if (_setup.getSkippedColumns() !=null &&
-                ((_setup.get_parse_columns_indices()==null) || (_setup.get_parse_columns_indices().length==0)))
+        if (_setup.getSkippedColumns() != null &&
+                ((_setup.get_parse_columns_indices() == null) || (_setup.get_parse_columns_indices().length == 0)))
           throw new H2OIllegalArgumentException("Parser:  all columns in the file are skipped and no H2OFrame" +
                   " can be returned."); // Need this to send error message to R
-
         p.parseChunk(in.cidx(), din, dout);
+        for (int partitionIndex = 0; partitionIndex < _partitionedBy.length; partitionIndex++) {
+          Matcher matcher = Pattern.compile(_partitionedBy[partitionIndex] + "=([^\\/\\\\]+)").matcher(in.vec()._key.toString());
+          matcher.find();
```

Comment:
should we check the result and provide understandable error?

Reply:
Draft. Will definitely be removed.